app/Http/Controllers/UsersController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Http\Requests;
+
+use App\User;
+
+class UsersController extends Controller
+{
+
+    public function index()
+    {
+        $users = User::all();
+        return view('users.index', compact('users'));
+    }
+
+    public function show($id)
+    {
+        $user = User::findOrfail($id);
```

Comment:
@mansoursaid001  So it should be like this right?
   
 public function show($id)
    {
        // Will return a ModelNotFoundException if no user with that id
        try
        {
            $user = User::findOrFail($id);
        }
        // catch(Exception $e) catch any exception
        catch(ModelNotFoundException $e)
        {
            // lists all available methods for exception object
            // Note that: dd($value) Dumps the $variable and end execution of the script.
            dd(get_class_methods($e))
            dd($e)
        }

        return view('users.show', compact('user'));
    }